ccm_web/server/src/invitation/cirrus-invitation.service.ts

```diff
@@ -0,0 +1,60 @@
+import { Injectable } from '@nestjs/common'
+import { ConfigService } from '@nestjs/config'
+import { Config } from '../config'
+import FormData from 'form-data'
+import { randomUUID } from 'crypto'
+import axios from 'axios'
+import { InvitationAPIError } from './invitation.errors'
+import { CanvasUserLoginEmail } from '../canvas/canvas.interfaces'
+import { HttpMethod } from '../api/api.utils'
+import baseLogger from '../logger'
+
+const logger = baseLogger.child({ filePath: __filename })
+
+@Injectable()
+export class CirrusInvitationService {
+  url: string
+  entityID: string
+  sponsorName: string
+  key: string
+  secret: string
+
+  constructor (private readonly configService: ConfigService<Config, true>) {
+    const invitationConfig = configService.get('invitation', { infer: true })
+    this.url = invitationConfig.apiURL
+    this.entityID = invitationConfig.apiEntityID
+    this.sponsorName = invitationConfig.apiSponsorName
+    this.key = invitationConfig.apiKey
+    this.secret = invitationConfig.apiSecret
+  }
+
+  async sendInvitations (users: CanvasUserLoginEmail[]): Promise<string> {
+    const userEmails: string[] = users.map(user => user.email)
+    const emailAddressCSV = `emailAddress\n${userEmails.join('\n')}`
+
+    const data = new FormData()
+    data.append('cfile', emailAddressCSV, 'fake_file_name.csv')
+    data.append('spEntityId', this.entityID)
+    data.append('sponsorEppn', this.sponsorName)
+    data.append('clientRequestID', 'ccm-' + randomUUID())
+
+    try {
+      const response = await axios({
+        method: HttpMethod.Post,
+        url: this.url,
+        auth: {
+          username: this.key,
+          password: this.secret
+        },
+        headers: { ...data.getHeaders() },
+        data: data
+      })
```

Comment:
Yes, we'd talked about this before in a Zoom meeting.  This and some other things took a low priority because I was trying to figure out the roles checking code.

Reply:
I'm working on switching from `axios` to `HttpModule` now, because I see that the way Axios works (or perhaps the way I'm using it), it masks incoming error messages.  The only message I ever see is `"Request failed with status code 400"`.

I see where `HttpModule` is referenced in `CanvasModule`, but no usage of it in `CanvasService`.  I'll dig around in the `canvas` hierarchy and see if I find its usage.

Reply:
It's probably likely that I don't understand TypeScript or perhaps NestJS imports well enough, but I don't see any explicit usage of `HttpModule` by code in the `canvas` hierarchy (aside from references in `CanvasModule`).  However, I do see that `HttpService` is imported and used in `CanvasService`.  Perhaps that's what you meant to tell me.

I think I can use it the same way in the invitation module.

Reply:
Yeah, sorry, you use `HttpService` from the `HttpModule`. `HttpModule` is registered in `CanvasModule`.

Reply:
> I'm working on switching from `axios` to `HttpModule`

@lsloan how long do you anticipate the switch will take ?

Reply:
I was hoping it would be pretty quick and would be done within a couple hours on Friday.  However, it was not as straightforward as I expected.

I began updating the invitation code to use `HttpModule`/`HttpService` as @ssciolla did for `CanvasService`/`CanvasService`.  The invitation code was approximately modeled on the Canvas one, so `CirrusInvitationModule` already had the `HttpModule` references in it.  I added code to `CirrusInvitationService` to make use of it instead of Axios.

I looked up documentation for `HttpModule` because the POST request I needed to make was different in two ways from other examples I see in our code.  The Cirrus API request requires:

1. Upload a file of data.  It seems that using `FormData` is a recommended technique for that with `HttpModule`, just as it was with Axios.
2. Use Basic Auth.  I tried using an `Auth` key in the config argument to `HttpModule` as I saw some example for.  Didn't seem to be working.  So, I changed the tactic to generate my own `Authorization` header.

At this point, I'm not sure whether it's actually working.  The call completes without error, but I'm not able to process the response as I expected to.

I'd appreciate some advice on this section of `CirrusInvitationService`: https://github.com/lsloan/ccm/blob/d66b3ed52ca56f4f97661d64a6d8b5a22b96aeff/ccm_web/server/src/invitation/cirrus-invitation.service.ts#L71-L81

Reply:
FWIW, if you think this change from Axios is taking to long, I can switch back to using it.  Sam had said he didn't require this change for merging this PR.

Reply:
This was resolved.